src/Form/EntryForm.php

```diff
@@ -22,52 +23,80 @@ public function getFormID() {
   /**
    * {@inheritdoc}
    */
-  public function buildForm(array $form, FormStateInterface $form_state) {
-    $account = $this->userStorage->load($form_state->get('uid'));
+  public function buildForm(array $form, FormStateInterface $form_state, AccountInterface $user = null) {
+    $tfaManager = \Drupal::service('tfa.manager');
+    $tfa = $tfaManager->getProcess($user);
 
-    //$tfa = tfa_get_process($account);
 
-    /*
     // Check flood tables.
-    if (_tfa_hit_flood($tfa)) {
-      \Drupal::moduleHandler()->invokeAll('tfa_flood_hit', [$tfa->getContext()]);
-      return drupal_access_denied();
-    }
-    */
+    //@TODO Reimplement Flood Controls.
+//    if (_tfa_hit_flood($tfa)) {
+//      \Drupal::moduleHandler()->invokeAll('tfa_flood_hit', [$tfa->getContext()]);
+//      return drupal_access_denied();
+//    }
+//
 
     // Get TFA plugins form.
-    //$form = $tfa->getForm($form, $form_state);
-    //if ($tfa->hasFallback()) {
-    $form['actions']['fallback'] = array(
-      '#type' => 'submit',
-      '#value' => t("Can't access your account?"),
-      '#submit' => array('tfa_form_submit'),
-      '#limit_validation_errors' => array(),
-      '#weight' => 20,
-    );
-    //}
+    $form = $tfa->getForm($form, $form_state);
+
+    //If there is a fallback method, set it.
+    if ($tfa->hasFallback()) {
+      $form['actions']['fallback'] = array(
+        '#type' => 'submit',
+        '#value' => t("Can't access your account?"),
+        '#submit' => array('tfa_form_submit'),
+        '#limit_validation_errors' => array(),
+        '#weight' => 20,
+      );
+    }
 
     // Set account element.
     $form['account'] = array(
       '#type' => 'value',
-      '#value' => $account,
+      '#value' => $user,
     );
 
-    return parent::buildForm($form, $form_state);
+    return $form;
   }
 
   /**
    * {@inheritdoc}
    */
   public function validateForm(array &$form, FormStateInterface $form_state) {
-    return parent::validateForm($form, $form_state);
+    $user = $form_state->getValue('account');
+    $tfaManager = \Drupal::service('tfa.manager');
+    $tfa = $tfaManager->getProcess($user);
+    $tfa->validateForm($form, $form_state);
   }
 
   /**
    * {@inheritdoc}
    */
   public function submitForm(array &$form, FormStateInterface $form_state) {
-    parent::submitForm($form, $form_state);
+    $user = $form_state->getValue('account');
```

Comment:
Can we put in some comments for story telling here? Not sure what all is going on